src/test/java/io/logz/jmx2graphite/GraphiteITest.java

```diff
@@ -0,0 +1,176 @@
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static io.restassured.RestAssured.given;
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SuppressWarnings({"SpellCheckingInspection", "rawtypes"})
+public class GraphiteITest {
+
+    private final static Logger logger = LoggerFactory.getLogger(GraphiteITest.class);
+
+    private static GenericContainer graphiteContainer;
+    private static Stopwatch beforeClassStopwatch;
+    public static final String TEST_SERVICE_HOST = "testhost";
+    public static final String TEST_SERVICE_NAME = "testservice";
+    private static File tempDir;
+
+    @BeforeClass
+    public static void beforeClass() {
+        beforeClassStopwatch = Stopwatch.createStarted();
+        tempDir = createTempDir();
+
+        graphiteContainer = new GenericContainer("sitespeedio/graphite:0.9.14")
+                .withExposedPorts(2003, 80)
+                .withFileSystemBind(tempDir.getAbsolutePath(),
+                                    "/var/log/carbon",
+                                    BindMode.READ_WRITE)
+                .withClasspathResourceMapping("carbon.conf",
+                                              "/opt/graphite/conf/carbon.conf",
+                                                BindMode.READ_ONLY)
+                .withClasspathResourceMapping("blacklist.conf",
+                                              "/opt/graphite/conf/blacklist.conf",
+                                                BindMode.READ_ONLY)
+                .withClasspathResourceMapping("storage-schemas.conf",
+                                              "/opt/graphite/conf/storage-schemas.conf",
+                                                BindMode.READ_ONLY)
+                .waitingFor(new HostPortWaitStrategy().withStartupTimeout(Duration.ofSeconds(10)))
+                .withLogConsumer(new Slf4jLogConsumer(logger).withPrefix("graphite"));
+
+        graphiteContainer.setPortBindings(Lists.newArrayList("2003:2003", "1180:80"));
+        graphiteContainer.start();
+        beforeClassStopwatch.stop();
+    }
+
+    private static File createTempDir() {
+        File tempDir = new File("/tmp/mytest"+System.currentTimeMillis());
+        if (!tempDir.mkdirs()) throw new RuntimeException("failed creating dir");
+        logger.info("Logging to {}", tempDir.getAbsolutePath());
+        return tempDir;
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        if (graphiteContainer != null) graphiteContainer.stop();
+        logger.info("Setup time took {} [sec]", beforeClassStopwatch.elapsed(TimeUnit.SECONDS));
+    }
+
+    @Test
+    public void sanity() {
+        createGraphiteClient();
+    }
+
+//    @Ignore("Remote this line to run")
+    @Test
+    public void testWrite() {
```

Comment:
We're actually testing it's ability to withstand restart of Graphite server, so maybe name method accordingly?